util/config/config.go

```diff
@@ -3,34 +3,56 @@ package config
 import (
 	"encoding/json"
 	"fmt"
-	"io/ioutil"
+	"io"
 	"net/http"
 	"os"
+	"text/template"
 
 	"github.com/kelseyhightower/envconfig"
 	"gopkg.in/yaml.v2"
 )
 
 func ReadConfig[Config interface{}](
-	path string, envBase string,
+	path string, templateData interface{}, envBase string,
 ) (*Config, error) {
 	file, err := os.Open(path)
 	if err != nil {
 		return nil, fmt.Errorf("failed to open config file: %v", err)
 	}
 	defer file.Close()
 
-	fileData, err := ioutil.ReadAll(file)
+	fileData, err := io.ReadAll(file)
 	if err != nil {
 		return nil, fmt.Errorf("failed to read config file: %v", err)
 	}
 
+	configTemplate, err := template.New("config").Parse(string(fileData))
+	if err != nil {
+		return nil, err
+	}
+
+	configReader, configWriter := io.Pipe()
+	templateErr := make(chan error)
+	go func() {
+		err := configTemplate.Execute(configWriter, templateData)
+		configWriter.Close()
+		templateErr <- err
+	}()
```

Comment:
Why did you want to do this in a goroutine? My initial reaction is this should be synchronous.

Reply:
Ahh I see the example on [io.Pipe](https://pkg.go.dev/io#Pipe)